src/commands/text.ts

```diff
@@ -0,0 +1,470 @@
+/*************************************************
+ * Abstract classes of text blocks
+ ************************************************/
+
+/**
+ * Blocks of plain text, with one or two TextPiece's as children.
+ * Represents flat strings of typically serif-font Roman characters, as
+ * opposed to hierchical, nested, tree-structured math.
+ * Wraps a single HTMLSpanElement.
+ */
+class TextBlock extends MQNode {
+  ctrlSeq = '\\text';
+  ariaLabel = 'Text';
+  replacedText?:string;
+  anticursorPosition?:number;
+
+  replaces (replacedText:Fragment | string) {
+    if (replacedText instanceof Fragment)
+      this.replacedText = replacedText.remove().jQ.text();
+    else if (typeof replacedText === 'string')
+      this.replacedText = replacedText;
+  };
+
+  jQadd (jQ:$) {
+    super.jQadd(jQ);
+    const endsL = this.ends[L];
+    if (endsL) {
+      const child = this.jQ[0].firstChild;
+      if (child) {
+        endsL.jQadd(child);
+      }
+    }
+    return this.jQ;
+  };
+
+  createLeftOf (cursor:Cursor) {
+    var textBlock = this;
+    super.createLeftOf(cursor);
+
+    cursor.insAtRightEnd(textBlock);
+
+    if (textBlock.replacedText)
+      for (var i = 0; i < textBlock.replacedText.length; i += 1)
+        textBlock.write(cursor, textBlock.replacedText.charAt(i));
+
+    const textBlockR = textBlock[R];
+    if (textBlockR) textBlockR.siblingCreated(cursor.options, L);
```

Comment:
```
 37 ⋮    │-    if (textBlock[R].siblingCreated) textBlock[R].siblingCreated(cursor.options, L);
 38 ⋮    │-    if (textBlock[L].siblingCreated) textBlock[L].siblingCreated(cursor.options, R);
 39 ⋮    │-    textBlock.bubble(function (node) { node.reflow(); });
    ⋮ 46 │+    const textBlockR = textBlock[R];
    ⋮ 47 │+    if (textBlockR) textBlockR.siblingCreated(cursor.options, L);
    ⋮ 48 │+    const textBlockL = textBlock[L];
    ⋮ 49 │+    if (textBlockL) textBlockL.siblingCreated(cursor.options, R);
    ⋮ 50 │+    textBlock.bubble(function (node) { node.reflow(); return undefined});
 40 ⋮ 51 │   };
```
The diff here makes me curious if we are sure that the siblingCreated method will always be defined for this class/node. It seems like if the old code was remotely safe we shouldn't need this check. It would seem odd to make the particular check for a method being present in the old code. Any thoughts @laughinghan ?

Reply:
I think this is one of those cases where `0` works but `undefined` doesn't. I think this was originally checking both:
1) textBlock[R] is defined
2) that block had the siblingCreated method on it.

If `textBlock[R] === 0` we can still look for the method on that number. But it won't exist. So we early return.

There's a chance we added a base noop `siblingCreated` method to all nodes. Then all we needed to check was that the node existed.